### PR TITLE
try out the cypress GH action

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves #NUMBER

**Overall change:**
_A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by karinathomasbbc** [comment]
Hey, totally lurking on this PR 👀, apologies! I wonder if GH actions are having the same issue as Travis, in that the cookie banner tests cannot work outside the EU? We may need to consider setting the `CYPRESS_SKIP_EU` env var like in Travis config (but have absolutely no idea where to do that here!) https://github.com/bbc/simorgh/blob/6ba0f194e7fba5e9cae4c32255ac6b5c2748702a/.travis.yml#L30

**Comment by chris-hinds** [comment]
> Hey, totally lurking on this PR 👀, apologies! I wonder if GH actions are having the same issue as Travis, in that the cookie banner tests cannot work outside the EU? We may need to consider setting the `CYPRESS_SKIP_EU` env var like in Travis config (but have absolutely no idea where to do that here!)
> 
> https://github.com/bbc/simorgh/blob/6ba0f194e7fba5e9cae4c32255ac6b5c2748702a/.travis.yml#L30

Just added it here https://github.com/bbc/simorgh/pull/7256/commits/84838061e885c7994955cb6ec94c2deebd50db6c lets see what happens!

**Comment by karinathomasbbc** [comment]
Worked! 😱 Next thing... does it cost us a fortune to run these GH actions?

**Comment by chris-hinds** [comment]
> Worked! 😱 Next thing... does it cost us a fortune to run these GH actions?

image

:)

At the moment no, they are free!!!!!!

Id like to try and get these running all of our CI tasks.

I currently have unit tests and integration tests on one action and cypress on another action and I like the separation.

We should try and break down that mammoth `npm run test:ci` cmd and run each item as a separate script in the main CI action

**Comment by chris-hinds** [comment]
Current running tests in the CI action

image

File: .github/workflows/simorgh_ci.yml [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
         run: |
           npm ci
-          npx apache2-license-checker
-          npm test
+          npm run build
+      - name: Unit Tests
+        run: npm run test:unit
+      - name: Integration Tests
+        run: npm run test:integration:ci
+      - name: Linting
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh_ci.yml]: could we move linting to before Unit tests - fail fast etc

File: .github/workflows/simorgh_ci.yml [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
+        run: npm run test:lint
+      - name: Puppeteer
+        run: |
+          npm i puppeteer@latest
+          npm run test:puppeteer
```
Comment (karinathomasbbc) [inline .github/workflows/simorgh_ci.yml]: We might need to do `npm run start` here too

**Pull request closed without merging**